src/service_daemon.rs

```diff
@@ -1430,12 +1437,48 @@ impl Zeroconf {
 
         buf.truncate(sz); // reduce potential processing errors
 
+        let is_unicast = !pktinfo.addr_dst.is_multicast();
+
+        // Ignore unicast packets outside the local link
+        if is_unicast {
+            let should_respond = match (pktinfo.addr_src.ip(), &intf_sock.intf.addr) {
```

Comment:
This actually has slightly different behaviour for IPv6 than my check. RFC states:

> Since it is possible
   for a unicast query to be received from a machine outside the local
   link, responders SHOULD check that the source address in the query
   packet matches the local subnet for that link (or, in the case of
   IPv6, the source address has an on-link prefix) and silently ignore
   the packet if not.

For IPv6, the `valid_ip_on_intf` function checks the subnet mask against the IP address, whereas I check if the address has an on-link prefix.

Reply:
You are right. Never mind. (I thought it covers the ipv6 case, but it's not)